### PR TITLE
Fix incorrect calculation of percentage for blacklisted outputs

### PR DESCRIPTION
**Pull request by Doy-lee**

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -8020,9 +8020,13 @@ void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>>
       uint64_t pick = first_rct + crypto::rand<uint64_t>() % n_rct;
 
       {
-        double percent_of_outputs_blacklisted = output_blacklist.size() / (double)n_rct;
+        float percent_of_outputs_blacklisted = static_cast<float>((output_blacklist.size() / (double)n_rct) * 100.0);
         if (static_cast<int>(percent_of_outputs_blacklisted + 1) > 5)
```
Comment (jagerman) [inline src/wallet/wallet2.cpp]: I think this could be simplified quite a lot to just:
```C++
if (output_blacklist.size() > 0.05 * n_rct) {
    ....
}
```

**Comment by jagerman** [comment]
While this PR fixes the calculation, I think there is a deeper issue here.  Looking at the current version of the code, it seems the only way this error can trigger is if:

`output_blacklist.size() / (double)n_rct`

is at least 5, which means the blacklist is already 5 times larger than n_rct.  With the new code that is going to be equal 500 (the correct percentage) instead, but that's certainly not going to avoid the error but rather trigger it more often.

**Comment by Doy-lee** [comment]
Errr, yeah that's not right fix. `n_rct / 100.0 * output_blacklist.size() > 5` or alternatively your one was the intention there. Good thing I sat on that to merge later and not on the spot.

**Comment by Doy-lee** [comment]
PR is moved to https://github.com/loki-project/loki/pull/539 since I updated the way to exclude blacklisted outputs.

**Pull request closed without merging**